src/main/java/org/opentripplanner/routing/api/request/StreetMode.java

```diff
@@ -1,140 +1,150 @@
+  BIKE_RENTAL(
+    Feature.ACCESS,
+    Feature.TRANSFER,
+    Feature.EGRESS,
+    Feature.WALKING,
+    Feature.CYCLING,
+    Feature.RENTING
+  ),
   /**
    * Walk to a scooter rental point, ride a scooter to a scooter rental drop-off point, and walk the
    * rest of the way. This can include scooter rental at fixed locations or free-floating services.
    */
-  SCOOTER_RENTAL(true, true, true, true, false, false, true, true, false, false),
+  SCOOTER_RENTAL(
+    Feature.ACCESS,
+    Feature.TRANSFER,
+    Feature.EGRESS,
+    Feature.WALKING,
+    Feature.SCOOTER,
+    Feature.RENTING
+  ),
   /**
    * Car only
    * <p>
    * Direct mode only.
    */
-  CAR(true, false, false, false, false, true, false, false, false, false),
+  CAR(Feature.ACCESS, Feature.DRIVING),
   /**
    * Start in the car, drive to a parking area, and walk the rest of the way.
    * <p>
    * Direct mode and access mode only.
    */
-  CAR_TO_PARK(true, false, false, true, false, true, false, false, true, false),
+  CAR_TO_PARK(Feature.ACCESS, Feature.WALKING, Feature.DRIVING, Feature.PARKING),
   /**
    * Walk to a pickup point along the road, drive to a drop-off point along the road, and walk the
    * rest of the way. This can include various taxi-services or kiss & ride.
    */
-  CAR_PICKUP(true, false, true, true, false, true, false, false, false, true),
+  CAR_PICKUP(Feature.ACCESS, Feature.EGRESS, Feature.WALKING, Feature.DRIVING, Feature.PICKUP),
   /**
    * Walk to a car rental point, drive to a car rental drop-off point and walk the rest of the way.
    * This can include car rental at fixed locations or free-floating services.
    */
-  CAR_RENTAL(true, true, true, true, false, true, false, true, false, false),
+  CAR_RENTAL(
+    Feature.ACCESS,
+    Feature.TRANSFER,
+    Feature.EGRESS,
+    Feature.WALKING,
+    Feature.DRIVING,
+    Feature.RENTING
+  ),
 
   /**
    * Using a car hailing app like Uber or Lyft to get to a train station or all the way to the destination.
    */
-  CAR_HAILING(true, false, true, false, false, true, false, false, false, true),
+  CAR_HAILING(Feature.ACCESS, Feature.EGRESS, Feature.DRIVING, Feature.PICKUP),
 
   /**
    * Encompasses all types of on-demand and flexible transportation.
    */
-  FLEXIBLE(true, false, true, true, false, false, false, false, false, false);
-
-  final boolean access;
-
-  final boolean transfer;
-
-  final boolean egress;
-
-  final boolean includesWalking;
-
-  final boolean includesBiking;
-
-  final boolean includesDriving;
+  FLEXIBLE(Feature.ACCESS, Feature.EGRESS, Feature.WALKING);
+
+  private enum Feature {
+    ACCESS,
+    EGRESS,
+    TRANSFER,
+    WALKING,
+    CYCLING,
+    DRIVING,
+    SCOOTER,
+    RENTING,
+    PARKING,
+    PICKUP,
+  }
 
-  final boolean includesScooter;
+  private final Set<Feature> features;
 
-  final boolean includesRenting;
+  StreetMode(Feature first, Feature... rest) {
```

Comment:
That's a nice way of making sure that you don't pass zero-lenght arrays.

Reply:
Btw a random note, is it normal in Java to have the `...` next to the type, not next to the field. To me this looks ugly after using other languages but I think prettier formatted it like this.

Reply:
This pattern is copied from the EnumSet constructor and make the integration with it simpler - I agree with @optionsome that it is not very pretty - but it is not part of the API of the class witch make it a bit simpler to accept. The underlying problem is limitations in Java´s reflection.